File: schemas/analytics.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


class StoreAnalytics(BaseModel):
    store_id: int
    store_address: str
    count: int
    total_items: int
    total_cost: float


class SupplyAnalyticsResponse(BaseModel):
    total_supplies: int
    total_items: int
    total_cost: float
    supplies_per_store: list[StoreAnalytics]


class StoreSalesAnalytics(BaseModel):
    store_id: int
    store_address: str
    count: int
    total_amount: float


class TopProduct(BaseModel):
    product_id: int
    name: str
    total_sold: int


class PaymentStats(BaseModel):
    method_name: str
    total_amount: float


class SalesAnalyticsResponse(BaseModel):
    total_sales: int
    total_amount: float
    average_check: float
    sales_per_store: list[StoreSalesAnalytics]
    top_product: Optional[TopProduct]
    payments_stats: list[PaymentStats]

File: frontend/src/api/analytics.ts
import api from './api'

const BASE_URL = "/admin"

export interface AnalyticsFilters {
    store_id?: number
    start_date?: string
    end_date?: string
}

//
// ----- ПОСТАВКИ -----
//

export interface StoreSupplyAnalytics {
    store_id: number
    store_address: string
    count: number
    total_items: number
    total_cost: number
}

export interface SupplyAnalyticsResponse {
    total_supplies: number
    total_items: number
    total_cost: number
    supplies_per_store: StoreSupplyAnalytics[]
}

export const getSupplyAnalytics = async (
    filters: AnalyticsFilters
): Promise<SupplyAnalyticsResponse> => {
    const response = await api.get<SupplyAnalyticsResponse>(`${BASE_URL}/analytics/supplies`, {
        params: filters,
    })
    return response.data
}

//
// ----- ПРОДАЖИ -----
//

export interface StoreSalesAnalytics {
    store_id: number
    store_address: string
    count: number
    total_amount: number
}

export interface TopProduct {
    product_id: number
    name: string
    total_sold: number
}

export interface PaymentStats {
    method_name: string
    total_amount: number
}

export interface SalesAnalyticsResponse {
    total_sales: number
    total_amount: number
    average_check: number
    sales_per_store: StoreSalesAnalytics[]
    top_product?: TopProduct
    payments_stats?: PaymentStats[]
}

export const getSalesAnalytics = async (
    filters: AnalyticsFilters
): Promise<SalesAnalyticsResponse> => {
    const response = await api.get<SalesAnalyticsResponse>(`${BASE_URL}/analytics/sales`, {
        params: filters,
    })
    return response.data
}